Fix level filters

clockvoid opened the pull request:
## Issue
- close #709 

## Overview (Required)
- rename AudienceCategory to Level
- create member level to Session model
- create LevelEntity
- fix FilterTest
- rename AudienceCategory to Level

日本語でこの変更を行った理由と私の思いについて書かせてください．
まず，AudienceCategoryという名前は去年のものだと思いますが，今年はLevelということで，視聴者のレベルではなく，セッションそのもののレベルであり，これは，セッションそのものの特性を指していることから，すべてのAudienceCategoryと名のついたものをLevelという名前に変えさせていただきました．
その際，iOS側のクラスの名前も変わってしまうことになります（なるはずですよね？）が，iOSではまだフィルターは実装されておらず，変更の効く範囲だと思いますので，このようにさせていただきました．
また，セッションのエンティティに`levels`という要素が加わったことにより，キャッシュのデータベースが変更されました．この事により，アプリのストレージをクリアしないとアップデート時に機能しなくなりますが，マイグレーションの処理は書いていません．理由としては，このアプリはまだPlayStoreに上がっておらず，開発段階であり，開発者の皆さまがアプリを再インストールすることで使えるようになるならば，スピード感を優先してマイグレーションはつけなくて良いと判断したためです．
最後に，テストについてです．今回私がいじったのは`model`モジュールですが，このモジュールの`jvmTest`モジュールはなぜかテストとして認識されていませんでした．`model/build.gradle`を見ると，`jvm()`の記述がなかったため，それを追加することにより`jvmTest`が動くようにしました．また，FilterTestに関しましては，今回の変更ですべてのパターンを調べると18通りとなり，非常に記述が長くなってしまうため，最小限ですべてのパターンを調べることができているように記述したつもりです．

## Links
- #614 

## Screenshot
Before | After
:--: | :--:
<img src="https://user-images.githubusercontent.com/6965122/73733843-04321b00-4780-11ea-8a33-6ad2257dd661.png" width="300" /> | <img src="https://user-images.githubusercontent.com/6965122/73733706-c8975100-477f-11ea-87da-53f27fd5c18a.png" width="300" />

takahirom commented:
アプリ内で実態にあった名前を変えるのはいいと思います！！👍👍

> jvmの記述がなかったため，それを追加することによりjvmTestが動くようにしました

👍👍👍👍

マイグレーションについてですが、現在マイグレーションが必要になると自動的にDBが破棄される設定になっています！スキーマのバージョンだけ上がっていれば大丈夫です！

clockvoid commented:
スキーマのバージョンを上げておりませんので，変更します

nashcft commented:
👍👍👍 

`jvmTest` についてですが、現状CIではテストが実行されていないので、`.circleci/config.yml` や `.github/workflows/on_release_branch.yml` のテストを実行している箇所に `jvmTest` タスクを追加すると良さそうに思いました 👀

clockvoid commented:
@nashcft 
I added `jvmTest` to `.circleci/config.yml` like [this](https://github.com/DroidKaigi/conference-app-2020/pull/711/commits/66169e2b469e0c9d39f17a74e20cd331ed654950#diff-1d37e48f9ceff6d8030570cd36286a61R67), but CircleCI's Gradle said `No cached version of com.soywiz.korlibs.klock:klock-jvm:1.8.6 available for offline mode.`.
In my environment, `./gradlew cleanAllTests testDebugUnitTest jvmTest --offline` is run and pass correctly. (Of course, I cleaned in advance.)
I am short on experience with CI so, I don't understand why it happens.
Please suggest me some solutions 🙏

clockvoid commented:
I think to create a cache with jvmTest dependencies when caching all of the other structures.
However, I can't find the function of caching dependencies for now...

takahirom commented on model/src/commonMain/kotlin/io/github/droidkaigi/confsched2020/model/Level.kt:
```diff
@@ -0,0 +1,14 @@
+package io.github.droidkaigi.confsched2020.model
+
+enum class Level(val rawValue: LocaledString) {
+    BEGINNER(LocaledString("初級", "Beginner")),
```
I think it is better to have a string such as "BEGINNER" as a property. `Level(val id :String, val rawValue: LocaledString) ` This is because this class is caused by the change of API response. The property name can be id. 👀 
For example, you have created a class called Level, but you cannot change it that way.

clockvoid replied:
I see.
I agree with you. I will fix it!

takahirom commented:
Thanks! Almost LGTM!

nashcft commented:
@clockvoid 
You're right, we should have cached dependencies for `jvmTest` but it seems we don't.
Resolving and downloading dependencies are executed [here](https://github.com/DroidKaigi/conference-app-2020/blob/bcd913a43c6949f2dbf1fefca913be573a5a8803/.circleci/config.yml#L28). I'm not sure but it seems that `dependencies` task doesn't download the listed dependencies unlike `androidDependencies` (and `androidDependenciesExtra`, which is extended version defined in this project).

takahirom commented:
It's better to have a cache, but after the merge. 👍

nashcft commented:
@takahirom 
I think it's OK but we have to remove `jvmTest` from CircleCI workflow before merge because no test is executed in this branch now.

takahirom commented:
It's a problem that the test doesn't run, but on another branch, we seem to have a problem with CI testing. 👀  ( https://github.com/DroidKaigi/conference-app-2020/pull/671 )
https://github.com/DroidKaigi/conference-app-2020/issues/714#issue-559762916

nashcft commented:
I see. Thanks for the information 🙏
